src/frontend/apps/desk/src/features/mail-domains/components/forms/CreateMailboxForm.tsx

```diff
@@ -0,0 +1,292 @@
+        duration: 4000,
+      });
+
+      setIsFormVisible(false);
+    },
+  });
+
+  const closeModal = () => setIsFormVisible(false);
+
+  const onSubmitCallback = (event: React.FormEvent) => {
+    event.preventDefault();
+    void methods.handleSubmit((data) =>
+      createMailbox({ ...data, mailDomainId: mailDomain.id }),
+    )();
+  };
+
+  return (
+    <FormProvider {...methods}>
+      <Modal
+        isOpen
+        leftActions={
+          <Button
+            color="secondary"
+            fullWidth
+            onClick={closeModal}
+            disabled={methods.formState.isSubmitting}
+          >
+            {t('Cancel')}
+          </Button>
+        }
+        onClose={closeModal}
+        closeOnClickOutside
+        hideCloseButton
+        rightActions={
+          <Button
+            color="primary"
+            fullWidth
+            type="submit"
+            form={FORM_ID}
+            disabled={methods.formState.isSubmitting}
+          >
+            {t('Submit')}
+          </Button>
+        }
+        size={ModalSize.MEDIUM}
+        title={
+          <Box $align="center" $gap="1rem">
+            <IconCreateMailbox
+              width={48}
+              color={colorsTokens()['primary-text']}
+              title={t('Mailbox creation form')}
+            />
+            <Text $size="h3" $margin="none" role="heading" aria-level={3}>
+              {t('Create a mailbox')}
+            </Text>
+          </Box>
+        }
+      >
+        <Box $width="100%" $margin={{ top: 'large', bottom: 'xl' }}>
+          {queryState.isError && (
+            <TextErrors className="mb-s" causes={queryState.error.cause} />
+          )}
+          {methods ? (
+            <Form
+              methods={methods}
+              t={t}
+              mailDomain={mailDomain}
+              onSubmitCallback={onSubmitCallback}
+            />
+          ) : null}
+        </Box>
+      </Modal>
+    </FormProvider>
+  );
+};
+
+const Form = ({
+  methods,
+  t,
+  mailDomain,
+  onSubmitCallback,
+}: {
+  methods: UseFormReturn<CreateMailboxParams>;
+  t: TFunction;
+  mailDomain: MailDomain;
+  onSubmitCallback: (event: React.FormEvent) => void;
+}) => (
+  <form onSubmit={onSubmitCallback} id={FORM_ID}>
+    <Box $direction="column" $width="100%" $gap="3rem" $margin="auto">
+      <Box $margin={{ horizontal: 'none' }}>
+        <Controller
+          control={methods.control}
+          name="first_name"
+          render={({ field, fieldState }) => (
+            <Input
+              aria-invalid={!!fieldState.error}
+              label={t('First name')}
+              state={fieldState.error ? 'error' : 'default'}
+              text={
+                fieldState.error ? t('Please enter your first name') : undefined
+              }
+              onBlur={field.onBlur}
+              onChange={(e) => methods.setValue(field.name, e.target.value)}
+              value={String(field.value)}
+            />
```

Comment:
What about this way, like that after an error if the user do a change, the error will not be displayed anymore ?
```suggestion
            <Input
              aria-invalid={!!fieldState.error}
              label={t('First name')}
              state={fieldState.error ? 'error' : 'default'}
              text={
                fieldState.error ? t('Please enter your first name') : undefined
              }
              {...methods.register('first_name')}
            />
```

Reply:
Small video to show the difference between `firstname` and `lastname` with this code, after an error, if the user type the error disappear directly:
[scrnli_5_29_2024_9-46-13 AM.webm](https://github.com/numerique-gouv/people/assets/25994652/2f0944fb-5f63-4687-b379-c6d00a85fe71)